src/Simulation/DepletionVoltage.jl

```diff
@@ -21,155 +21,138 @@ end
-calculate_electric_potential!(sim)
-estimate_depletion_voltage(sim, 2, 1600:1:2500)
-```
+SolidStateDetectors.estimate_depletion_voltage(sim, 2, field_sim_settings = (verbose = true,))
 
-!!! warn
-    This method only works if the initial `sim` was calculated for a case in which the detector is fully depleted.
+```
 
 !!! note
     The accuracy of the result depends on the precision of the initial simulation.
     
 See also [`is_depleted`](@ref).
 """
-function estimate_depletion_voltage(sim::Simulation{T}, contact_id::Int,
-            potential_range::AbstractRange = range(extrema(broadcast(c -> c.potential, sim.detector.contacts))..., length = 1001);
-            tol = T(1e-2),
-            verbose = true)::T where {T <: AbstractFloat}
-    
-    @assert is_depleted(sim.point_types) "This method only works for fully depleted simulations. Please increase the potentials in the configuration file to a greater value."
-    @assert first(potential_range) * last(potential_range) >= 0 "Please search for depletion voltage either in a fully positive or a fully negative range. Currently you were looking in the range
-        from $(first(potential_range)) to $(last(potential_range))"
-    #potential_range = (last(potential_range) > 0 && first(potential_range) >= 0) || (step(potential_range) < 0) ? potential_range : reverse(potential_range)
-    
-    if verbose
-        @info "Looking for the depletion voltage applied to contact $(contact_id) "*
-              "in the range $(extrema(potential_range).*u"V") in steps of $(step(potential_range)*u"V")."
-    end
-    
-    for c in sim.detector.contacts
-        if c.potential == 0 && c.id != contact_id continue end
-        if ismissing(sim.weighting_potentials[c.id]) || sim.weighting_potentials[c.id].grid != sim.electric_potential.grid
-            @info "The weighting potential $(c.id) need to be defined on the same grid as the electric potential. Adjusting weighting potential $(c.id) now."
-            _adapt_weighting_potential_to_electric_potential_grid!(sim, c.id)
-        end
-    end
-    
-    # ϕρ is the electric potential resulting only from the impurity density
-    # and setting all potentials on the contacts to zero
-    ϕρ = deepcopy(sim.electric_potential.data)
-    for c in sim.detector.contacts
-        if c.potential == 0 continue end
-        ϕρ .-= c.potential * sim.weighting_potentials[c.id].data
-    end
-    
-    inside = findall(p -> p & bulk_bit > 0 && p & update_bit > 0, sim.point_types.data)
-    
-    ϕmin, ϕmax = extrema((ϕρ .+ potential_range[1] * sim.weighting_potentials[contact_id].data)[inside])
-    eps = 1e-3 * sign( potential_range[length(potential_range)÷2] )
-    initial_depletion::Bool = ϕmax - ϕmin < maximum((abs(potential_range[1]), eps))
-    depletion_voltage::T = NaN
-    start_local_search::T = 0
+function estimate_depletion_voltage(
+        sim::Simulation{T}, 
+        contact_id::Int;
+        field_sim_settings::NamedTuple = (verbose = false,)
```

Comment:
This doesn't seem to work well, if you wanna change verbose from the outside.
I guess this is done to set the default value of verbose to `false`?

Is this intended that one needs to know the name of the keyword argument and pass the settings as `NamedTuple`?

Reply:
It's supposed to allow for changing any field simulation settings. 
Verbose was only chosen as default to restrict the argument to NamedTuples.
The `merge` call will overwrite preset settings defined in the function body.

I guess we could turn it into an argument with default value instead.